lib/services/sshAdapter.js

```diff
@@ -0,0 +1,64 @@
+"use strict";
+
+var Bluebird = require("bluebird");
+
+function SshAdapter (ec2) {
+	var SSH_POLLING_INTERVAL = 30; /*Time in Seconds*/
+
+	this.checkIfSshable = Bluebird.method(function (instanceID) {
+		return this.canSsh(instanceID)
+		.then(function (sshable) {
+			if (sshable) {
+				return Bluebird.resolve();
+			}
+			else {
+				return Bluebird.delay(SSH_POLLING_INTERVAL * 1000)
+				.then(this.checkIfSshable.bind(this, instanceID));
+			}
+		}.bind(this));
+	});
+
+	this.startSshPolling = Bluebird.method(function (instanceID) {
+
+		return this.checkIfSshable(instanceID)
+		.catch(function (err) {
+			return Bluebird.reject(err);
+		});
+	});
+
+	this.canSsh = Bluebird.method(function (instanceID) {
+		var sshable;
+		var params = {
+			InstanceIds : [ instanceID ]
+		};
+
+		return ec2.describeInstanceStatusAsync(params)
+		.then(function (data) {
+			sshable = false;
```

Comment:
It looks like this variable is only used in this promise frame. Is there a reason that it is scoped to the entire method?